### ISSUE TITLE
Environment variable input validation on CHPL_*_COMPILER

### ISSUE THREAD

**Issue by BryantLam** [chapel-lang/chapel#12824]
Low priority. Environment variables need input validation. I started a build process with:
```terminal
CHPL_HOST_COMPILER=clang
CHPL_TARGET_COMPILER=gcc # <-- This is wrong and should be gnu.
CHPL_LLVM=system # LLVM 8
```
The build kicked off but broke around the build for runtime/libchplmalloc.a:
```terminal
...
***** src/timers/generic/ *****
***** src/qio/auxFilesys/curl/ *****
***** src/qio/auxFilesys/hdfs/ *****
***** ./ *****
cp ../build/runtime/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/./src/main.o ../lib/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/main.o
ar -rc ../lib/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/libchplmalloc.a ./../build/runtime/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/src/chpl-mem-replace-malloc.o
../lib/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/libchplmalloc.a # <--- This line!
make[3]: execvp: ../lib/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/libchplmalloc.a: Permission denied.
make[3]: *** [../lib/linux64/gcc/x86_64/arch-unknown/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-locks/hwloc/re2/fs-none/lib_pic-none/libchplmalloc.a] Error 127
make[2]: *** [all.helpme] Error 2
make[1]: *** [runtime] Error 2
make: *** [comprt] Error 2
```
The line that broke is trying to execute a static library because Makefile variable `$(RANLIB)` wasn't set by `make`.

As a minor point, runtime/Makefile.help should echo `$(RANLIB)` in printstuff.helpme.

**Comment by bradcray** [comment]
Hi Bryant — When I set CHPL_TARGET_COMPILER=gnu and do a `make`, I get a bunch of messages like:

```
Warning: Unknown compiler: "gcc"
```

Do you not?

**Comment by BryantLam** [comment]
It does not emit any warnings for me. Searching through the build logs doesn't show any signs that anything is wrong; the build just kicks off like normal.

Interestingly, if I set both to invalid `CHPL_{HOST,TARGET}_COMPILER=gcc`, the build still kicks off but detects that I have `clang++` in my system LLVM and uses that (until it breaks later at the same point in the parent post).

**Comment by bradcray** [comment]
Ah, OK thanks.  I'm seeing now with a bit more poking that this warning is due to my CPU settings.  I'll take a quick look on what it would take to emit a similar warning more generally...

**Comment by bradcray** [comment]
I've got a sketch of a fix for this in #12837.  It both generates a warning when a CHPL_*_COMPILER setting is unfamiliar and makes the RANLIB stuff a bit more bulletproof.
